php-practice_2.php

```diff
@@ -1,15 +1,103 @@
 <?php
 // Q1 tic-tac問題
 
+echo "1から100までのカウントを開始します" . "\n";
 
+for ($i = 1; $i <= 100; $i++) {
+  if ($i % 4 === 0 && $i % 5 === 0) {
+    echo "tic-tac\n";
+  } elseif($i % 4 === 0) {
+    echo "tic\n";
+  } elseif($i % 5 === 0) {
+    echo "tac\n";
+  } else {
+    echo $i."\n";
+  }
+}
 // Q2 多次元連想配列
+$personalInfos = [
+  [
+      'name' => 'Aさん',
+      'mail' => 'aaa@mail.com',
+      'tel'  => '09011112222'
+  ],
+  [
+      'name' => 'Bさん',
+      'mail' => 'bbb@mail.com',
+      'tel'  => '08033334444'
+  ],
+  [
+      'name' => 'Cさん',
+      'mail' => 'ccc@mail.com',
+      'tel'  => '09055556666'
+  ],
+];
 
+var_dump($personalInfos);
+var_dump($personalInfos[1]['name']);
+var_dump($personalInfos[1]['tel']);
+
+// 問題1
+echo $personalInfos[1]['name'] ."の電話番号は" . $personalInfos[1]['tel'] . "です。";
+
+// 問題2
+$counter = 1;
+
+foreach ($personalInfos as  $y) {
+  echo $counter . "番目の" . $y['name'] . "さんのメールアドレスは" . $y['mail'] . "で、電話番号は" . $y['tel'] . "です。\n";
+  $counter++;
+}
+
+// 問題3
+$ageList = [25, 30, 18];
+
+foreach ($personalInfos as $index => $info) {
+  $personalInfos[$index]['age'] = $ageList[$index];
+}
+
+var_dump($personalInfos);
 
 // Q3 オブジェクト-1
+class Student
+{
+    public $studentId;
+    public $studentName;
 
+    public function __construct($id, $name)
+    { throw new Exception("学籍番号は正の整数です。");
+        if (!is_int($id) || $id <= 0) {
+
+        }
+        $this->studentId = $id;
+        $this->studentName = $name;
+    }
```

Comment:
Q3
[ask]

- この`__constract`メソッド内でやっていることを説明してみて欲しいです！

Reply:
IDが整数かつ正の数であることを確認していて、
その条件を満たさないのであれば　"学籍番号は正の整数です。”　というエラーを出します。
条件を満たした$id,$nameをstudentId、studentNameに代入しています。

Reply:
ちゃんと理由を持ってコードが書けていて素晴らしいですね！👍
ただ、そうするとなると、下記のように修正が必要そうですね！

①if文の中にthrow文を書く。
（そうしないと、「条件を満たさない場合にエラーを出す」が実現しない）

②代入処理は、else文で繋げて条件がfalseだった時の処理として書く。
（そうしないと、「整数かつ正の数」でなかった時、if文内に書いたthrowが実行されないだけで、if文の後に書いた処理は普通に順番に実行されるので、「整数かつ正の数」でなかった時も代入されてしまう）


```suggestion
    public function __construct($id, $name)
    {
        if (!is_int($id) || $id <= 0) {
              throw new Exception("学籍番号は正の整数です。");
        } else {
              $this->studentId = $id;
              $this->studentName = $name;
        }
    }
```